example/lib/app_central.dart

```diff
@@ -1,54 +1,60 @@
 import 'package:flutter_chess_board/flutter_chess_board.dart';
 import 'package:universal_chess_driver/central.dart';
+import 'package:universal_chess_driver/peripherial.dart';
+
+class AppCentralRound implements CentralRound {
+  String? _variant;
+  String? _fen;
+  String? _lastMove;
+
+  AppCentralRound(this._variant, this._fen, this._lastMove);
+
+  String? get variant => _variant;
+  String? get fen => _fen;
+  String? get lastMove => _lastMove;
+}
 
 class AppCentral implements Central {
   ChessBoardController _chessController;
+  Peripherial? _peripherial = null;
 
   AppCentral(this._chessController);
 
-  String get fen => _chessController.getFen();
-  String get variant => "standard";
-  String? get lastMove => _lastMove();
+  List<String> get features => ["msg", "last_move"];
+  List<String> get variants => ["standard"];
+  AppCentralRound get round =>
+      AppCentralRound("standard", _chessController.getFen(), _lastMove());
 
-  Future<bool> move(String uci) {
-    return Future.value(_chessController.makeMoveUci(uci: uci));
+  @override
+  void onPeriherialConnected(Peripherial p) {
+    _peripherial = p;
   }
 
-  Future<bool> isUnspefiedPromotion(String uci) {
-    var srcPiece = _chessController.game.get(uci.substring(0, 2));
-    if (srcPiece?.type.name != "p") return Future.value(false);
-
-    if ((_chessController.game.turn == Chess.WHITE &&
-            uci[1] == "7" &&
-            uci[3] == "8") ||
-        (_chessController.game.turn == Chess.BLACK &&
-            uci[1] == "2" &&
-            uci[3] == "1"))
-      return Future.value(true);
-    else
-      return Future.value(false);
+  @override
+  void onPeriherialDisconnected() {
+    _peripherial = null;
   }
 
-  Future<String> obtainPromotedPawn() {
-    return Future.value("q");
-  }
+  @override
+  void onPeripheralRoundChange() {}
 
-  void indicateOutOfSync(String peripherialFen) {
-    print("Peripherial is out of sync: $peripherialFen");
+  @override
+  void onPeripheralMove(String uci) {
+    bool isRejected = !_chessController.makeMoveUci(uci: uci);
```

Comment:
just `isAccepted`, to not have inverse logic?

Reply:
fixed